python/lsst/daf/persistence/swiftStorage.py

```diff
@@ -0,0 +1,519 @@
+            The object representing the connection. The object may or may not
+            actually be connected.
+        """
+        user = os.getenv('SWIFT_USERNAME')
+        if user is None:
+            raise RuntimeError(
+                'SwiftStorage could not find SWIFT_USERNAME in environment')
+        key = os.getenv('SWIFT_PASSWORD')
+        if key is None:
+            raise RuntimeError(
+                'SwiftStorage could not find SWIFT_PASSWORD in environment')
+
+        # TODO are auth_version and self._version (from the input URI, usually
+        # 'v2.0') supposed to correlate? How?
+
+        return self.swift.Connection(
+            authurl=self._url, user=user, key=key,
+            tenant_name=self._tenantName, auth_version=2)
+
+    def containerExists(self):
+        """Query if the container for this SwiftStorage exists."""
+        try:
+            self._connection.head_container(self._containerName)
+        except self.swift.ClientException:
+            return False
+        return True
+
+    def deleteContainer(self):
+        """Delete all the objects in this container"""
+        respHeaders, containers = self._connection.get_account()
+        for container in containers:
+            name = container['name']
+            if name == self._containerName:
+                headers, objects = self._connection.get_container(
+                    self._containerName)
+                for obj in objects:
+                    self._connection.delete_object(self._containerName, obj['name'])
+                self._connection.delete_container(self._containerName)
+                break
+
+    def write(self, butlerLocation, obj):
+        """Writes an object to a location and persistence format specified by
+        ButlerLocation
+
+        This file uses PosixStorage to write the object to a file on disk (that
+        is to say: serialize it). Then the file is uploaded to the swift
+        container. When we have better support for pluggable serializers,
+        hopefully the first step of writing to disk can be skipped and the
+        object can be serialzied and streamed directly to the swift container.
+
+        Parameters
+        ----------
+        butlerLocation : ButlerLocation
+            The location & formatting for the object to be written.
+        obj : object instance
+            The object to be written.
+        """
+        swiftLocations = butlerLocation.getLocations()
+        # Here the ButlerLocation is modified sligtly to write to a temporary
+        # file via PosixStorage. (Then the temporary file is written to the
+        # swift container)
+        localFile = tempfile.NamedTemporaryFile()
+        butlerLocation.locationList = [localFile.name]
+        butlerLocation.storage = PosixStorage('/')
+        butlerLocation.storage.write(butlerLocation, obj)
+        butlerLocation.locationList = swiftLocations
+        with open(localFile.name, 'rb') as f:
```

Comment:
Can you avoid re-opening the file? If think `tempfile` warns that opening temporary file second time it is non-portable (and there are other reasons to avoid that). I think you can just do `localFile.seek(0)` and then read everything from it.